### ISSUE TITLE
cant build on node v12.3.1

### ISSUE THREAD

**Issue by japrescott** [SonicHedgehog/detect-character-encoding#18]
node: v12.3.1
node-gyp: 3.8.0
npm: 6.9.0

```

/root/.node-gyp/12.3.1/include/node/v8.h:3359:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/root/.node-gyp/12.3.1/include/node/v8config.h:322:3: note: in definition of macro 'V8_DEPRECATE_SOON'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../icuWrapper.cpp:67:96: warning: 'bool v8::Object::Set(v8::Local<v8::Value>, v8::Local<v8::Value>)' is deprecated: Use maybe version [-Wdeprecated-declarations]
  obj->Set(Nan::New<v8::String>("confidence").ToLocalChecked(), Nan::New<v8::Number>(confidence));
                                                                                                ^
In file included from /root/.node-gyp/12.3.1/include/node/v8-internal.h:14:0,
                 from /root/.node-gyp/12.3.1/include/node/v8.h:25,
                 from /root/.node-gyp/12.3.1/include/node/node.h:63,
                 from ../icuWrapper.cpp:1:
/root/.node-gyp/12.3.1/include/node/v8.h:3359:26: note: declared here
                     bool Set(Local<Value> key, Local<Value> value));
                          ^
/root/.node-gyp/12.3.1/include/node/v8config.h:322:3: note: in definition of macro 'V8_DEPRECATE_SOON'
   declarator __attribute__((deprecated(message)))
   ^~~~~~~~~~
../icuWrapper.cpp: In function 'void Init(v8::Local<v8::Object>)':
../icuWrapper.cpp:75:72: error: no matching function for call to 'v8::FunctionTemplate::GetFunction()'
   Nan::New<v8::FunctionTemplate>(DetectCharacterEncoding)->GetFunction());
                                                                        ^
In file included from /root/.node-gyp/12.3.1/include/node/node.h:63:0,
                 from ../icuWrapper.cpp:1:
/root/.node-gyp/12.3.1/include/node/v8.h:5947:46: note: candidate: v8::MaybeLocal<v8::Function> v8::FunctionTemplate::GetFunction(v8::Local<v8::Context>)
   V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(
                                              ^~~~~~~~~~~
/root/.node-gyp/12.3.1/include/node/v8.h:5947:46: note:   candidate expects 1 argument, 0 provided
icuWrapper.target.mk:126: recipe for target 'Release/obj.target/icuWrapper/icuWrapper.o' failed
make: *** [Release/obj.target/icuWrapper/icuWrapper.o] Error 1
make: Leaving directory '/mystuff/node_modules/detect-character-encoding/build'
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/root/.nvm/versions/node/v12.3.1/lib/node_modules/npm/node_modules/npm-lifecycle/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Linux 4.15.0-45-generic
gyp ERR! command "/root/.nvm/versions/node/v12.3.1/bin/node" "/root/.nvm/versions/node/v12.3.1/lib/node_modules/npm/node_modules/npm-lifecycle/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /mystuff/node_modules/detect-character-encoding
gyp ERR! node -v v12.3.1
gyp ERR! node-gyp -v v3.8.0

```

**Comment by hybloid** [comment]
Same thing.

**Comment by Lohardt** [comment]
Here for the same reason. For now I will roll back to an older node version for the current project.

**Comment by sonicdoe** [comment]
Thanks for reporting this 👍 I’ve just released v0.8.0 which fully supports Node.js v12.
